### PR TITLE
Use the same avatar colour when creating 1:1 DM rooms

### PR DESCRIPTION
**Pull request by weeman1337**
closes https://github.com/vector-im/element-web/issues/23476

Contains `filterConsole` tweak. Should be reviewed commit-wise.

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Use the same avatar colour when creating 1:1 DM rooms ([\#9850](https://github.com/matrix-org/matrix-react-sdk/pull/9850)). Fixes vector-im/element-web#23476.

### CODE REVIEW

File: test/components/views/avatars/__snapshots__/RoomAvatar-test.tsx.snap [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      style="font-size: 23.400000000000002px; width: 36px; line-height: 36px;"
+    >
+      D
+    </span>
+    <img
+      alt=""
+      aria-hidden="true"
+      class="mx_BaseAvatar_image"
+      data-testid="avatar-img"
+      src="data:image/png;base64,00"
```
Comment (weeman1337) [inline test/components/views/avatars/__snapshots__/RoomAvatar-test.tsx.snap:21]: Unfortunately this does not assert the rendered colour. But at least the text content is different per test: `D`, `L` and `T`.

**Comment by gsouquet** [comment]
Should also fix https://github.com/vector-im/element-web/issues/20957

**Comment by weeman1337** [comment]
> Should also fix [vector-im/element-web#20957](https://github.com/vector-im/element-web/issues/20957)

I don't think so. This only targets „local rooms“. Other behaviour should be unchanged.

File: test/test-utils/console.ts [hunk 2]
```diff
@@ -26,11 +26,11 @@ const originalFunctions: FilteredConsole = {
...
 /**
  * Allows to filter out specific messages in console.*.
+ * Automagically restores the original function by implementing an afterAll hook.
  *
  * @param ignoreList Messages to be filtered
- * @returns function to restore the console
  */
-export const filterConsole = (...ignoreList: string[]): (() => void) => {
+export const filterConsole = (...ignoreList: string[]): void => {
     for (const [key, originalFunction] of Object.entries(originalFunctions)) {
```
Comment (SimonBrandner) [inline test/test-utils/console.ts:34]: Would it make sense to wrap this in `beforeEach()` or `beforeAll()`?

**Review by SimonBrandner — APPROVED** [review]
LGTM

File: test/test-utils/console.ts (hunk 2 above)
Comment (weeman1337) [reply test/test-utils/console.ts:34]: Clean up is done in `afterAll` → going to put this into a `beforeAll` :+1:

**Pull request merged**